File: examples/step01/tests/timer.lib/elapsed.cc
// -*- C++ -*-
// -*- coding: utf-8 -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2026 all rights reserved

// support
#include <iostream>
// the library
#include <timer/Timer.h>


int
main()
{
    timer::Timer t;
    t.start();
    // burn enough cycles that steady_clock records non-zero elapsed time
    volatile int dummy = 0;
    for (int i = 0; i < 1'000'000; ++i) {
        dummy += i;
    }
    t.stop();
    // the elapsed time must be strictly positive
    if (t.elapsed() <= 0.0) {
        std::cerr << "FAILED: elapsed time should be positive after start()/stop()" << std::endl;
        return 1;
    }
    // all done
    return 0;
}


// end of file

File: examples/step02/tests/timer.pkg/elapsed.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2026 all rights reserved

# support
import sys
import time

# the package
from timer import Timer


def test() -> int:
    """
    Verify that elapsed time is positive after a start/stop cycle.
    """
    t = Timer()
    t.start()
    # sleep long enough that perf_counter records non-zero elapsed time
    time.sleep(0.01)
    t.stop()
    # the elapsed time must be strictly positive
    if t.elapsed <= 0.0:
        print(
            "FAILED: elapsed time should be positive after start()/stop()",
            file=sys.stderr,
        )
        return 1
    # all done
    return 0


# main
if __name__ == "__main__":
    status = test()
    raise SystemExit(status)


# end of file
